src/ranch_conns_sup.erl

```diff
@@ -285,8 +299,19 @@ terminate(#state{shutdown=Shutdown}, Reason, NbChildren) ->
 			erlang:send_after(Shutdown, self(), kill)
 	end,
 	wait_children(NbChildren),
+	update_stats(terminate, StatsCounters, Id, NbChildren),
 	exit(Reason).
 
+update_stats(Type, StatsCounters, Id) ->
+	update_stats(Type, StatsCounters, Id, 1).
+
+update_stats(accept, StatsCounters, Id, N) ->
+	%% Accepts are counted in the odd indexes.
+	counters:add(StatsCounters, 2*Id-1, N);
+update_stats(terminate, StatsCounters, Id, N) ->
+	%% Terminates are counted in the even indexes.
+	counters:add(StatsCounters, 2*Id, N).
```

Comment:
Make a separate function for each. Maybe like `inc_accept` and `inc_terminate`. I'm not sure the default `1` value is necessary, `inc_accept(Counters, Id, 1)` is perfectly understandable.

Reply:
Fine with me.

Reply:
also done